### PR TITLE
apiserver: ALWAYS return apiserver.UpgradeInProgress

### PR DESCRIPTION
**Pull request by davecheney**
Always, _always_, return apiserver.UpgradeInProgress if there is an
upgrade in progress.

(Review request: http://reviews.vapour.ws/r/4070/)

### CODE REVIEW

File: apiserver/admin_test.go [hunk 1]
```diff
@@ -527,7 +527,7 @@ func (s *loginSuite) TestLoginValidationDuringUpgrade(c *gc.C) {
 		c.Assert(err, jc.ErrorIsNil)
 
 		err = st.APICall("Client", 1, "", "DestroyModel", nil, nil)
-		c.Assert(err, gc.ErrorMatches, ".*upgrade in progress - Juju functionality is limited.*")
+		c.Assert(errors.Cause(err), gc.DeepEquals, &rpc.RequestError{Message: "upgrade in progress", Code: ""})
```
Comment (davecheney) [inline apiserver/admin_test.go]: This still isn't right, the Code should be "upgrade in progress", not the message. But at least it's coming in an rpc.RequestError.

**Comment by cherylj** [comment]
Please hold off on merging this until after we ship beta2 (should be in the next day or so).

**Comment by davecheney** [comment]
Will do, that'll give me more time for manual testing.

**Comment by davecheney** [comment]
$$merge$$

**Comment by davecheney** [comment]
$$merge$$

**Pull request merged**